pyfa_tool/lib_functions.py

```diff
@@ -8,189 +8,486 @@
 
     Parameters
     ----------
     fa_filepath : Str
         Path to the FA-file.
-    fieldname : TYPE, optional
-        Name of the spatial field to extract from the FA file. The default is 'SFX.T2M'.
-    target_crs : Str, optional
-        EPSG identifier for the target CRS. The data will be reprojected to this if needed. The default is 'EPSG:4326'.
+    tmpdir : str, optional
+        The path to a temporary storage location for the json files. If None,
+        a new temporary folder is created. The default is None.
+    rm_tmpdir : bool, optional
+        Wether to remove the temporary storage direction, with all its content (= json
+        files) at the end of this function. The default is True.
 
     Returns
     -------
-    data : Xarray.DataArray
-        A DataArray containing the data in the target_crs. Meta data + CRS info is added to the data.attrs.
-
-    """
-    
-    json_path, fields_json_path, tmpdir = _Fa_to_json(fafile=fa_filepath,
-                                              fieldname=fieldname)
-    
-
-    
-    # =============================================================================
-    # check if json file is created
-    # =============================================================================
-    
-    if not os.path.isfile(json_path):
-        print(f'{fieldname} not found in {fa_filepath}.')
-    
-        # print available fields
-        fieldsdf = IO.read_json(jsonpath=fields_json_path,
-                                to_dataframe=True)
-    
-        if fieldsdf.shape[0] > 100:
-            fieldsdf = fieldsdf[0:100]
-            print(
-                f'There are {fieldsdf.shape[0]} stored in the {fa_filepath}. Here are the first 100 fields:')
-        else:
-            print(f'There are {fieldsdf.shape[0]} stored in the {fa_filepath}:')
-        print(fieldsdf)
-        print('To get all fields, use the get_fields() function.')
-    
-    
-        #TODO: define errors
-        shutil.rmtree(tmpdir)
-        # sys.exit(f'{fieldname} not found in {fa_filepath}.')
-        return None
-    
-    # =============================================================================
-    # Make xarray from json
-    # =============================================================================
-   
-    reproj_bool = True
-    
-    data = to_xarray.json_to_rioxarray(json_path=json_path,
-                                       reproject=reproj_bool,
+    fielddata : pandas.DataFrame
+        Available fields information in a Dataframe.
+
+    """
+    if not IO.check_file_exist(fa_filepath):
+        sys.exit(f'{fa_filepath} is not a file.')
+
+    # create at tmpdir if not provided
+    if tmpdir is None:
+        tmpdir = IO.create_tmpdir(location=os.getcwd())
+
+    # Run Rscript to generete a json file with all info
+    r_script = os.path.join(main_path, 'modules', 'rfa_scripts', 'get_fieldnames.R')
+    subprocess.call(["/usr/bin/Rscript", r_script, fa_filepath, tmpdir])
```

Comment:
When testing on the VSC the /usr/bin/Rscript does not seem to work.
```bash
pyfa.get_fieldnames('/dodrio/scratch/projects/2022_200/project_output/RMIB-UGent/vsc46032_kobe/PyFa/tests/data/ICMSHABOF+0288')
```

```error
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/dodrio/scratch/projects/2022_200/project_output/RMIB-UGent/vsc46032_kobe/PyFa/pyfa_tool/lib_functions.py", line 276, in get_fieldnames
    subprocess.call(["/usr/bin/Rscript", r_script, fa_filepath, tmpdir])
  File "/readonly/dodrio/apps/RHEL8/zen2-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/subprocess.py", line 345, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/readonly/dodrio/apps/RHEL8/zen2-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/subprocess.py", line 966, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/readonly/dodrio/apps/RHEL8/zen2-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/subprocess.py", line 1842, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/usr/bin/Rscript'
```

@vergauwenthomas  Maybe the idea was to use the _get_rbin function?

Reply:
Good fix!